lib/orbit-common/cache.js

```diff
@@ -154,6 +158,23 @@ var Cache = Class.extend({
     if (this.trackRevLinks && (op === 'add' || op === 'replace')) {
       this._addRevLinks(path, value, normalizedOperation);
     }
+
+    return true;
+  },
+
+  _isOperationRequired: function(operation){
+    if(operation.path[2] === '__rel'){
+      var recordExists = this.retrieve([operation.path[0], operation.path[1]]);
+      var valueMatches = this._valuesEqual(this.retrieve(operation.path), operation.value);
+      return recordExists && !valueMatches;
+    }
+
+    return true;
+  },
+
+  _valuesEqual: function(a, b){
+    if(a == null && b == null) return true;
```

Comment:
I added it because remove operations have a value of null and this.retrieve() returns undefined. I could add something a bit more explicit for the remove operations though.